homeassistant/components/yardian/coordinator.py

```diff
@@ -0,0 +1,78 @@
+"""Update coordinators for Yardian."""
+
+from __future__ import annotations
+
+import asyncio
+from dataclasses import dataclass
+import datetime
+import logging
+
+from pyyardian import AsyncYardianClient, NetworkException, NotAuthorizedException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, MANUFACTURER
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = datetime.timedelta(seconds=30)
+
+
+@dataclass
+class YardianDeviceState:
+    """Data retrieved from a Yardian device."""
+
+    zones: list[list]
+    active_zones: set[int]
+
+
+class YardianUpdateCoordinator(DataUpdateCoordinator[YardianDeviceState]):
+    """Coordinator for Yardian API calls."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        controller: AsyncYardianClient,
+    ) -> None:
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=entry.title,
+            update_method=self._async_update_data,
+            update_interval=SCAN_INTERVAL,
```

Comment:
If the result of the returned data can be compared with __eq__ set always_update=False to avoid dispatching the same update to entities